Architecture.md

```diff
@@ -28,7 +28,29 @@
 
 #### Client applications deployment
 
-***TODO: [INFRA] fill for the architecture used to deploy clients' applications***
+Pomegranate, the application and deployment manager, is in charge of monitoring and managing the containers' lifecycle, as well as configuring them.
+
+The [Rust](https://www.rust-lang.org/) programming language is used to develop the service. We used it because
+we wanted an efficient and reliable language that could be trusted with this job thanks to its compiled nature and the restrictions it brings to drive us to write better code,
```

Comment:
This sentence is too long and complex. The notion is good, but it has to be broken down and explained better. Also, *how* exactly do the restrictions help to write better code?

Reply:
Explaining all the things the compiler does to ~~torture~~ care after us would be a long task.
Perhaps I should link to an article that dives into the details ?

Reply:
Explaining a bit more some details could be great, and I do approve of the idea of inserting a link to a detailed article on this topic. Please ensure the source is trustworthy beforehand.

Reply:
For some reason I couldn't find a good article at all

Reply:
I broke down the sentences but didn't really try to elaborate a lot on the compiler. Nonetheless a new review is needed